### ISSUE TITLE
The reproduction effect of the Hayao dataset significantly differs from the effects of the provided weights.

### ISSUE THREAD

**Issue by TANQIanQ** [ptran1203/pytorch-animeGAN#30]
I trained the model using the default parameters.
python3 train.py --anime_image_dir dataset/Hayao \
                --real_image_dir dataset/train_photo \
                --model v2 \                 # animeGAN version, can be v1 or v2
                --batch 8 \
                --amp \                      # Turn on Automatic Mixed Precision training
                --init_epochs 10 \
                --exp_dir runs \
                --save_interval 1 \
                --gan_loss lsgan \           # one of [lsgan, hinge, bce]
                --init_lr 1e-4 \
                --lr_g 2e-5 \
                --lr_d 4e-5 \
                --wadvd 300.0\               # Aversarial loss weight for D
                --wadvg 300.0\               # Aversarial loss weight for G
                --wcon 1.5\                  # Content loss weight
                --wgra 3.0\                  # Gram loss weight
                --wcol 30.0\                 # Color loss weight
                --use_sn\                    # If set, use spectral normalization, default is False

torch==2.0.0+cu118
torchvision== 0.15.1+cu118
color-transfer-py==0.0.4

The effects obtained differ greatly from the effects of the weights you provided.
To achieve similar effects, do I need to change the parameters or is there an issue with my environment?
![1734417393383](https://github.com/user-attachments/assets/7bc2ad3f-4c02-4df3-b1a7-0cb53f3b6864)
 The left figure below shows my experimental results.

color_transfer_py  0.0.4
torch              1.13.0+cu116
torchaudio         0.13.0+cu116
torchvision        0.14.0+cu116
![1734428020081](https://github.com/user-attachments/assets/b4fc0cd9-a8cd-4021-b7cf-d98b8960b31a)
 The left figure below shows my experimental results.

**Comment by LuckyFrances** [comment]
我也出现了这样的问题！您的问题最后解决了吗？

**Comment by TANQIanQ** [comment]
没有，但是我猜测可能应该是训练集大小有关，
在这个项目的readme表示AnimeGANv2     由Google Landmark v2数据集进行训练。我当时仅仅使用了原始的AnimeGAN数据集



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2025年03月27日 22:39 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [ptran1203/pytorch-animeGAN] The reproduction effect of the Hayao dataset significantly differs from the effects of the provided weights. (Issue #30) |

我也出现了这样的问题！您的问题最后解决了吗？

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

LuckyFrances left a comment (ptran1203/pytorch-animeGAN#30)

我也出现了这样的问题！您的问题最后解决了吗？

**Comment by LuckyFrances** [comment]
But I went to look at the original project (which is the TensorFlow version), and the dataset seems to be no different from this PyTorch version. Where did readme mention the Google Landmark v2 dataset?

**Comment by ptran1203** [comment]
Sorry to hear that. I will answer your questions:

1. The generated images look different from the repo results.
- Yes, I also experience some unstable results sometimes. In my experiences, generated images change significantly over epochs (For example, epoch 100 may produce good results, while epoch 101 may not). So I have a `--save_interval` argument where you can see the generated image and checkpoint in each `save_interval` epoch. Based on that, you can pick the checkpoint that works best for you.

2. Why did I mention Google Landmark v2.
The Google Landmark v2 is just a different dataset. I replaced the original dataset in V2. The image quality of this dataset is better and most contains landscape images, which I think will improve result quality (Generate better at higher resolution)
